cg/clients/freshdesk/freshdesk_client.py

```diff
@@ -26,33 +27,42 @@ def __init__(self, base_url: str, api_key: str, order_email_id: int, env: str):
         self.env = env
         self.session = self._get_session()
 
-    def create_ticket(self, ticket: TicketCreate) -> TicketResponse:
+    @handle_client_errors
+    def create_ticket(self, ticket: TicketCreate, attachments: list[Path] = None) -> TicketResponse:
         """Create a ticket."""
-        LOG.debug(ticket.model_dump_json())
-        try:
+        LOG.info(ticket.model_dump_json(exclude_none=True))
+
+        # Prepare the ticket fields
+        ticket_data = ticket.model_dump(exclude_none=True)
+
+        if attachments:
+            # Multipart form-data
+            files = [
+                ("attachments[]", (attachment.name, open(attachment, "rb")))
+                for attachment in attachments
+            ]
+            response: Response = self.session.post(
+                url=self._url(EndPoints.TICKETS), data=ticket_data, files=files  # Form fields
+            )
+        else:
+            # JSON payload
             response: Response = self.session.post(
-                url=self._url(EndPoints.TICKETS),
-                json=ticket.model_dump(exclude_none=True),
+                url=self._url(EndPoints.TICKETS), json=ticket_data  # JSON data
```

Comment:
I'd remove this `_url` helper method, it just makes the code more difficult to follow

Reply:
👍